### PR TITLE
sap_ swpm: Prevent task "SAP SWPM Deployment - Finished" from failing if sap_swpm_sid or sap_swpm_pas_instance_nr not defined.

### PR DESCRIPTION
**Pull request by rob0d**
This happens when using advanced mode (e.g. Diag Agent installation) and sap_swpm_* variables are not defined as they are not needed. These two are used to display the status of SWPM after it finished and can fail the whole playbook even though SWPM finished correctly.

### CODE REVIEW

File: roles/sap_swpm/tasks/post_install.yml [hunk 1]
```diff
@@ -26,8 +26,8 @@
       - '   SAP SWPM deployment successfully completed '
       - ' '
       - '   SAP Product        -       {{ sap_swpm_product_catalog_id }}  '
-      - '   SID                -       {{ sap_swpm_sid }} '
-      - '   Primary Instance   -       {{ sap_swpm_pas_instance_nr }} '
+      - '   SID                -       {{ sap_swpm_sid | d("") }} '
+      - '   Primary Instance   -       {{ sap_swpm_pas_instance_nr | d("") }} '
```
Comment (Wabri) [inline roles/sap_swpm/tasks/post_install.yml:30]:
```suggestion
      - '   Primary Instance   -       {{ sap_swpm_pas_instance_nr | d('') }} '
```

File: roles/sap_swpm/tasks/post_install.yml [hunk 2]
```diff
@@ -26,8 +26,8 @@
       - '   SAP SWPM deployment successfully completed '
       - ' '
       - '   SAP Product        -       {{ sap_swpm_product_catalog_id }}  '
-      - '   SID                -       {{ sap_swpm_sid }} '
-      - '   Primary Instance   -       {{ sap_swpm_pas_instance_nr }} '
+      - '   SID                -       {{ sap_swpm_sid | d("") }} '
```
[the remark this answers is not in the document]
Comment (Wabri) [reply roles/sap_swpm/tasks/post_install.yml:29]:
```suggestion
      - '   SID                -       {{ sap_swpm_sid | d('') }} '
```

[the remark this answers is not in the document]
Comment (rob0d) [reply roles/sap_swpm/tasks/post_install.yml:29]: @Wabri @berndfinger I don't think we can use `d('')` as it's already inside single quotes. Options would be to escape the inside single quotes, change the outside single quotes to " (not consistent with the rest) or use just `d()`. The last option seems to work, shall I change it to that?

[the remark this answers is not in the document]
Comment (berndfinger) [reply roles/sap_swpm/tasks/post_install.yml:29]: > @Wabri @berndfinger I don't think we can use `d('')` as it's already inside single quotes.

You are right, it wouldn't work. Looking at
https://github.com/sap-linuxlab/community.sap_install/blob/42af6baab0310ec9c47994147d877f3d11ca4108/roles/sap_hana_install/tasks/post_install.yml#L192, we are also using single quotes for `msg:`, so please ignore my comment. In other words, let's use the single quotes for the outer level and stick with your proposal to use the double quotes for the inner level.

File: roles/sap_swpm/tasks/post_install.yml (hunk 1 above)
Comment (berndfinger) [reply roles/sap_swpm/tasks/post_install.yml:30]: @Wabri As discussed above, using nested quotes of the same time will not work. So we'll leave the double quotes here.

**Review by berndfinger — APPROVED** [review]
LGTM

File: roles/sap_swpm/tasks/post_install.yml (hunk 1 above)
Comment (Wabri) [reply roles/sap_swpm/tasks/post_install.yml:30]: Yes, I mark this as resolve

**Review by Wabri — APPROVED** [review]

**Pull request merged**